waku/common/message_hash.go

```diff
@@ -0,0 +1,39 @@
+package common
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+)
+
+// MessageHash represents an unique identifier for a message within a pubsub topic
+type MessageHash string
```

Comment:
I think this is not strictly correct. A message hash is `array[byte, 32]`
I'd happier if we rename it a little ;P
```suggestion
type MessageHashHexStr string
```

Reply:
I think this might be confusing for the user. A message hash is an `array[byte, 32]` for nwaku and at the protocol level.

For the end user of `waku-go-bindings`, a hash is that specific type and that's it, no matter which type we define it to. The end user shouldn't care or know if the hash is a hex string or an array. I don't think we should add more complex names in an user facing type just so it's consistent with the types we use under the hood.

I believe everything should be named in the simplest and most intuitive way for the user, no matter what we do under the hood

Reply:
ah I see your point and I 100% agree about simplicity.
In this particular case, naming this as `MessageHash` is misleading because a message hash is `array[byte, 32]`, and not a string. In fact, the type that is being defined in this line is a representation of msg_hash.

Nevertheless, go ahead with that. I won't block by this tiny detail :)

Reply:
Hahaha got it! Yes, it might be misleading for us, but not for the user, and this is a user-facing type.
It's basically a tradeoff, we sacrifice a bit of consistency with the naming in our internal implementation for a better user experience.

I don't think we should pass implementation specific complexities to the user. The user shouldn't know or care what's the type of the hash in the actual Waku protocols